client-v2/src/main/java/com/clickhouse/client/api/Client.java

```diff
@@ -460,4 +479,12 @@ private static Set<String> createFormatWhitelist(String shouldSupport) {
         }
         return Collections.unmodifiableSet(formats);
     }
+
+    private static final String INTERNAL_OPERATION_ID = "operationID";
+
+    private String startOperation() {
+        String operationId = UUID.randomUUID().toString();
```

Comment:
Are we sending operationId also as **query_id** this can be a good connection between our code and the actual operation in ClickHouse

Reply:
Good point.